src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -118,26 +122,29 @@ const CustomFieldInputSingleChoice = forwardRef(function CustomFieldInputSingleC
     const newValue = event.target.value;
 
     if (newValue === '') {
-      setValue(undefined);
-      setSearchValue(undefined);
+      clear();
       return;
     }
 
     setSearchValue(newValue);
     setShowOptions(true);
   }
 
-  useImperativeHandle(ref, () => ({
+  useImperativeHandle(selfRef, () => ({
     id: props.id,
     get value() {
-      return value;
+      return value ? [Number(value.id)] : [];
```

Comment:
Non-blocker but future concern:

I was hoping we would keep symmetry between the `value` prop and `value` ref APIs. I think the current custom field choice server APIs are wrong -- we can't do server side search if the `value` is a list of IDs. Keeping symmetry would also make it easier for developers to grok the `value`s: hey, it's the same as the thing we passed in. 

But we can just keep changing the `prop` and `ref` APIs as we improve the system overall.

Reply:
I agree on all counts, and think we should address this once we have a chance to really sit down and compile/enact changes on the API side, especially if it's us owning a new API endpoint special made for Custom Apps.